### PR TITLE
Remove spaces from DNs in SearchRequest

### PR DESCRIPTION
**Pull request by rkaw92**
This change fixes how DNs are serialized into BER for SearchRequest messages.
It drops spaces around commas, since it breaks some LDAP servers that expect
the DN to be separated by comma only as per RFC 4514. The old format of
spaced separators from LDAPv2 is thus no longer used.

For #611

### CODE REVIEW

File: lib/messages/search_request.js [hunk 1]
```diff
@@ -105,7 +105,13 @@ SearchRequest.prototype._parse = function (ber) {
 SearchRequest.prototype._toBer = function (ber) {
   assert.ok(ber)
 
-  ber.writeString(this.baseObject.toString())
+  // Format only with commas, since that is what RFC 4514 mandates.
+  // There's a gotcha here: even though it's called baseObject,
+  //  it can be a string or a DN object.
+  var formattedDN = dn.DN.isDN(this.baseObject)
```
Comment (jsumners) [inline lib/messages/search_request.js:111]: I think we should fork `node-asn1` into the org after publishing `ldapjs@2` and potentially clean up ambiguous things like this.

**Comment by jsumners** [comment]
@UziTech do you have any objections to this change?

File: lib/messages/search_request.js (hunk 1 above)
Comment (rkaw92) [reply lib/messages/search_request.js:111]: @jsumners Does `node-asn1` actually have anything to do with this? It's not a bad idea in general, just missing the idea as it regards DN formatting.

Comment (jsumners) [reply lib/messages/search_request.js:111]: It's the module that provides BER encoding.

**Comment by jsumners** [comment]
⚠️ This issue has been locked due to age. If you have encountered a recent
problem that seems to be covered by this issue, please open a new issue.

Please include a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example)
when opening a new issue.

**Pull request merged**